### PR TITLE
Add timestamp output

### PR DESCRIPTION
**Pull request by saleel**
## Motivation

Addresses #134 

## Change Summary

Add timestamp output to QR verifier circuit - It extracts the time form signed data and convert to Unix timestamp.

- Add a generic date string to timestamp convertor.
- Use that on extracted QR data in the main QR_verify circuit
- Add tests for timestamp convertor, and Aadhaar extraction.

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [change set](https://github.com/privacy-scaling-explorations/anon-aadhaar/CHANGELOG.md)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bug fix, or chore)
- [ ] PR includes documentation if necessary.

## Additional Context

- This PR is based to `feat/extractor-nullifier`. It should be changed to `develop` once #139 is merged.

### CODE REVIEW

**Review by Meyanis95 — APPROVED** [review]
LGTM

File: packages/circuits/helpers/timestamp.circom
Comment (vuvoth) [inline packages/circuits/helpers/timestamp.circom:79]: I think out is underconstaint

Comment (vuvoth) [reply packages/circuits/helpers/timestamp.circom:79]: A out signal don't have any constraint with input signal

Comment (saleel) [reply packages/circuits/helpers/timestamp.circom]: Are you sure? I thought it is constrained as the variable used is computed from the input signals. If I use any non-quadratic operation in computing variable which is later assigned to a signal, I will get compilation error. 
Anyway I have made some changes in the circuit to also address another issue. Can you check now?

Comment (vuvoth) [reply packages/circuits/helpers/timestamp.circom]: If we assign signal(or an expression contains signal) to variable, the circom can compile it. 
circom check non-quadratic operation for the left hand side and right hand side is signal I think.

File: packages/circuits/helpers/timestamp.circom
Comment (vuvoth) [inline packages/circuits/helpers/timestamp.circom:22]: I think we should change to: 
signal year <== numStringToNum(in[0]) * 1000 + numStringToNum(in[1]) * 100 + numStringToNum(in[2]) * 10 + numStringToNum(in[3])

numStringToNum must be component also. 
Because with current implement `year` is independence with signal `in[0], in[1], in[2], in[3]` . 

Same with month and day.

Comment (vuvoth) [reply packages/circuits/helpers/timestamp.circom:22]: Maybe I wrong! I need check it again

Comment (saleel) [reply packages/circuits/helpers/timestamp.circom:22]: I did think about it - but from what I understood, since the variable is assigned to a signal later in the circuit  essentially making the input (transformed) bind to the signal, this is ok? 
We dont necessarily want year to be dependent on the signal but the values in daysPassed should be.
I could be wrong - I will change it if you think this is under-constrained. I checked with `circom inspect` earlier and didn't see any warnings on these.

Comment (vuvoth) [reply packages/circuits/helpers/timestamp.circom:22]: I did think about it - but from what I understood, since the variable is assigned to a signal later in the circuit essentially making the input (transformed) bind to the signal, this is ok?
> I recheck it! I think you are correct. But I still worry about this.

Comment (saleel) [reply packages/circuits/helpers/timestamp.circom:22]: I think I will change it, just to be safe. Number of constrains are same either way.

File: packages/circuits/circuits/qr_verify.circom
Comment (saleel) [inline packages/circuits/circuits/qr_verify.circom:65]: @vuvoth I have removed the `reveal_timestamp` input and is always outputting formatted timestamp. Do you think we need to a conditional on this?
If we do, I think this should be a template param, so we can skip the conversion all together.

File: packages/circuits/circuits/extractor.circom
Comment (saleel) [inline packages/circuits/circuits/extractor.circom:83]: @vuvoth This is being done in QR_Verify circuit.

File: packages/circuits/circuits/qr_verify.circom
Comment (vuvoth) [reply packages/circuits/circuits/qr_verify.circom:65]: We don't the way developer use timestamp in their application. It's good if we still keep it like the option.

File: packages/circuits/circuits/extractor.circom
Comment (vuvoth) [reply packages/circuits/circuits/extractor.circom:83]: I think this should be done in extractor, because this circuit care about extract information.

**Review by vuvoth — APPROVED** [review]
LGTM! We can improve after merge.

**Pull request merged**
